images/win/scripts/Installers/Install-JavaTools.ps1

```diff
@@ -50,26 +50,49 @@ function Set-JavaPath {
 function Install-JavaFromAdoptOpenJDK {
     param(
         [string] $JDKVersion,
-        [string] $DestinationPath
+        [string] $Architecture = "x64"
     )
 
-    $assets = Invoke-RestMethod -Uri "https://api.adoptopenjdk.net/v3/assets/latest/$JDKVersion/hotspot"
-    $downloadUrl = ($assets | Where-Object {
+    # Get Java version from adopt openjdk api
+    $assetUrl = Invoke-RestMethod -Uri "https://api.adoptopenjdk.net/v3/assets/latest/${JDKVersion}/hotspot"
+    $asset = $assetUrl | Where-Object {
         $_.binary.os -eq "windows" `
-        -and $_.binary.architecture -eq "x64" `
+        -and $_.binary.architecture -eq $architecture `
         -and $_.binary.image_type -eq "jdk"
-    }).binary.package.link
+    }
+    $downloadUrl = $asset.binary.package.link
+    $fullJavaVersion = $asset.version.semver
 
+    # Download and extract java binaries to temporary folder
     $archivePath = Start-DownloadWithRetry -Url $downloadUrl -Name $([IO.Path]::GetFileName($downloadUrl))
-    Extract-7Zip -Path $archivePath -DestinationPath $DestinationPath
+    $javaTempPath = Join-Path -Path $env:TEMP -ChildPath "Java_$fullJavaVersion"
+    Extract-7Zip -Path $archivePath -DestinationPath $javaTempPath
+    $javaTempBinariesPath = Join-Path -Path $javaTempPath -ChildPath "\jdk*\"
+
+    # Create directories in toolcache path
+    $javaToolcachePath = Join-Path -Path $env:AGENT_TOOLSDIRECTORY -ChildPath "Java_Adoptium_jdk"
+    $javaVersionPath = Join-Path -Path $javaToolcachePath -ChildPath $fullJavaVersion
+    $javaArchPath = Join-Path -Path $javaVersionPath -ChildPath $architecture
```

Comment:
styling nitpick: follow the case for param variables `architecture` -> `Architecture`.
The same for all other code